src/Auth0RestClient.js

```diff
@@ -50,6 +50,36 @@ class Auth0RestClient {
     };
   }
 
+  _request(method, args) {
+    if (!this.options.includeResponseHeaders) {
+      return this.restClient[method](...args);
+    }
+
+    let callback;
+    // Handle the case where the promise variant is called without any args
+    // client.get() -> client.get({}, callback)
+    if (!args || !args.length) {
+      args = [{}];
+    }
+    // Handle the case where an undefined placeholder is defined for callback
+    // client.get(params, undefined) -> client.get(params, callback)
+    if (typeof args[args.length - 1] === 'undefined') {
+      args.pop();
+    }
+    if (typeof args[args.length - 1] === 'function') {
+      callback = args.pop();
+    }
+    return new Promise((resolve, reject) => {
+      this.restClient[method](...args, (err, data, headers) => {
```

Comment:
Here we are always using the `restClient` (`rest-facade`)'s callback method because this is the only one that returns both the body of the response and the headers